packages/cli/src/commands/source/catalog/new.ts

```diff
@@ -0,0 +1,57 @@
+import {SourceType} from '@coveord/platform-client';
+import {Command} from '@oclif/core';
+import {green} from 'chalk';
+import dedent from 'ts-dedent';
+import {
+  IsAuthenticated,
+  Preconditions,
+} from '../../../lib/decorators/preconditions';
+import {Trackable} from '../../../lib/decorators/preconditions/trackable';
+import {withSourceVisibility} from '../../../lib/flags/sourceCommonFlags';
+import {AuthenticatedClient} from '../../../lib/platform/authenticatedClient';
+
+export default class SourceCataloghNew extends Command {
+  public static description =
+    'Create a new catalog source in a Coveo organization';
+
+  public static flags = {
+    ...withSourceVisibility(),
+  };
+
+  public static args = [
+    {
+      name: 'name',
+      description: 'The name of the source to create.',
+      required: true,
+    },
+  ];
+
+  @Trackable()
+  @Preconditions(IsAuthenticated())
```

Comment:
Should there be a privilege check for "can create push source?"

Reply:
Good catch. I will address that in CDX-932